### PR TITLE
[RFC][language] Implement a primitive LibraTime module

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Implement a LibraTime module for a better wrapper of the notion of time on chain.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

**Comment by runtian-zhou** [comment]
General thoughts: 
1. Should we have a LibraTime.MutT vs LibraTime.T so that LibraTime.T is immutable?
2. Should we make a LibraTime.T a resource?
3. What are the other basic apis we need to support?

**Comment by huitseeker** [comment]
cc @zekun000 @dmitri-perelman just to keep an eye on Move's time implem

File: language/stdlib/modules/libra_time.mvir [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module LibraTime {
```
Comment (sblackshear) [inline language/stdlib/modules/libra_time.mvir]: To your top-level comment: I think we perhaps make this immutable and then call it `Timestamp` or `LibraTimestamp`. A mutable representation of a timestamp seems certain to cause confusion...

File: language/stdlib/modules/libra_time.mvir [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        *(&mut move(t).timestamp) = move(new_time);
+        return;
+    }
+
+    public add_millis(t: &mut Self.T, elapsed: u64) {
+        *(&mut move(t).timestamp) =  *(&copy(t).timestamp) + move(elapsed);
+        return;
+    }
+
+    public is_before(lhs: &Self.T, rhs: &Self.T): bool {
```
Comment (sblackshear) [inline language/stdlib/modules/libra_time.mvir]: - Perhaps `is_before_exclusive` to make it explicit that this is exclusive of the `rhs`?
- A variant of this where `rhs` is a `u64` is probably useful as well.
- We would probably also want `is_after_exclusive` (or similar)

**Comment by sblackshear** [comment]
> Should we make a LibraTime.T a resource?

My vote would be no; this doesn't need to be published at the top level, and there's no problem with copying it.

> What are the other basic apis we need to support?

I think utilities like `is_before_minutes(&Time.T, u64)`, `is_before_hours(&Time.T, u64)`, `is_before_days(&Time.T, u64)`, etc. would be useful.

**Comment by davidiw** [comment]
Please re-base and re-request if still valid.

**Pull request closed without merging**